csharp/src/Ice/InputStream.cs

```diff
@@ -2157,14 +2173,13 @@ public Enumerator(InputStream ins, InputStreamReader<T> read, int size)
 
             public bool MoveNext()
             {
-                if (_pos >= _size)
+                if (++_pos > _size)
                 {
```

Comment:
Maybe set pos to size + 1 here, for the unlikely event where lots of calls to MoveNext overflow pos?